feat/HNG-17-success-page-for-magic-link

oyin-systems opened the pull request:
<!-- Do not delete this PR template. Just edit it to include the required information -->

# Description

<!-- If your PR fixes an open issue, use `Closes #999` to link your PR with the issue. #999 stands for the issue number you are fixing -->

<!-- Github Issue Example: Closes #31 -->

**Closes #17**

# Changes proposed

## What were you told to do?
Title: Authentication Page -> Success page for magic link #17
Acceptance Criteria:
Responsive across all screen sizes (desktop, tablet and mobile)
Write tests for your code.
Use components where necessary (header, footer, buttons, modal and so on)
<!-- Write the title of the issue/feature you are working on -->

## What did you do?
I created a new page (Success.tsx)  and made it available as a route. I also had to create header and footer components since it was needed to complete the page. Then I performed tests according to the instructions.

<!-- Talk about the things you did eg. files changes, dependencies installed e.t.c -->

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

<!-- Mark all the applicable boxes. To mark the box as done follow the following conventions -->

<!--
[x] - Correct; marked as done
[X] - Correct; marked as done

[ ] - Not correct; marked as **not** done
-->

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos
Desktop
![desktop](https://github.com/user-attachments/assets/901c11ef-fb87-4c09-9240-13f25ae7326a)
Tablet
![tablet](https://github.com/user-attachments/assets/1d8cceab-9aed-4f18-a5d8-85d97397cd8e)
Mobile
![mobile](https://github.com/user-attachments/assets/4fba9d94-5fac-4538-8f9d-1b860f67f355)

<!-- If the changes are static page changes or UI changes add screenshots -->
<!-- If the changes involve implementing a functionality or working with apis, include a video
detailing how to implement the functionality and the request to the api and responses from the api endpoint-->
<!-- Add all the screenshots/videos which support your changes i.e before your change and after your change -->

# Link to Figma File
[Design](https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=2-938&t=p2gECcZQmYqvvrn7-4)

psalm987 commented:
the endpoint of the page should be /login/email-link/success according to the routing page, not /magic-link

oyin-systems commented:
> the endpoint of the page should be /login/email-link/success according to the routing page, not /magic-link

Yes, it is. the magic-link is the old one and I've deleted it. I already changed it to point to auth/login/email-link/success. Check the screenshot below:
![abcd](https://github.com/user-attachments/assets/97f80808-aa48-4a1e-84eb-b834b63f4881)

shalomtaiwo reviewed (approved):
Good.
